src/include/miopen/adam/problem_description.hpp

```diff
@@ -77,14 +71,8 @@ struct MIOPEN_INTERNALS_EXPORT ProblemDescription : ProblemDescriptionBase
           foundInfDesc(foundInfDesc_),
           stepInDesc(stepInDesc_),
           stepOutDesc(stepOutDesc_),
-          step(step_),
-          lr(lr_),
-          beta1(beta1_),
-          beta2(beta2_),
-          weight_decay(weight_decay_),
-          eps(eps_),
           amsgrad(amsgrad_),
-          maximize(maximize_),
+          correct_bias(correct_bias_),
```

Comment:
Yeah, now I see that the same Problem is used for both Adam and AdamW. Which means that these two are basically the same primitive. So let's add one more Solver to the existing Adam solvers, make sure that IsApplicable in the old Adam solver is updated (these should return false for the "adamw" Adam problems) and let the core of MIOpen to do the rest (i.e. select the proper Solver).

Reply:
If combining into one solver, it would be better if https://github.com/ROCm/MIOpen/pull/2957 was reviewed and merged first.